vuepress/docs/next/tutorials/create/ms/add-access-controls.md

```diff
@@ -1,109 +1,161 @@
+This confines use of the `getConference` method to users who are assigned either the `conference-user` or the `conference-admin` role on the Keycloak client configured for the microservice. 
 
-Now we should verify this security check is working.
+> Note: In local testing, the default client is `internal`. Refer to the [Spring Security documentation](https://spring.io/projects/spring-security) for more information.
 
-2. Start up your Keycloak, tableWidget MFE, and microservice. See [these instructions](./run-local.md) if you need a refresher but these are the basic commands using the ent CLI and Docker for keycloak.
-```
+### Step 2: Run your project in a local developer environment
+The following commands leverage the [ent CLI](../../../docs/reference/entando-cli.md).
+
+1. Start up your Keycloak instance
+``` sh
 ent prj ext-keycloak start
-ent prj be-test-run
 ```
-Using a separate cmdline:
+2. Initialize the tableWidget MFE
+``` sh
+ent prj be-test-run
 ```
+3. Use a new cmd line to start the microservice
+``` sh
 ent prj fe-test-run
 ```
+> Note: Refer to the [Run Blueprint-generated Microservices and Micro Frontends in Dev Mode tutorial](./run-local.md) for details.
+
+### Step 3: Access the tableWidget MFE
+
+1. In your browser, go to <http://localhost:3000>. This is typically the location of the tableWidget MFE.
+2. Access the tableWidget MFE with the default credentials of `username: admin`, `password: admin` 
 
-3. Access the tableWidget MFE, typically on <http://localhost:3000>, using the default admin/admin account. 
+> Note: Once authenticated, the message "No conferences are available" is generated. If you check your browser
+console, you should see a `403 (Forbidden)` error for the request made to `localhost:8080/services/conference/api/conferences`. This is expected because the admin user has not yet been granted the new role. 
 
-Once authenticated, you'll get the message "No conferences are available" and, if you check your browser console, you should see a `403 (Forbidden)` error for the request made to `localhost:8080/services/conference/api/conferences`. This is expected because we have not yet granted the new role to the admin user. 
+### Step 4: Login to Keycloak
 
-Now let's give the admin user the correct role. 
+1. Go to <http://localhost:9080> 
+2. Login using the the default credentials of `username: admin`, `password: admin`
 
-4. Login to keycloak on <http://localhost:9080> using the `admin/admin` credentials. 
+### Step 5: Create the `conference-user` and `conference-admin` roles 
 
-First we need to create the two roles per our requirements. We're going to add the roles to the `internal` client because it's the one configured by default in the Spring Boot application.yml.
```

Comment:
I feel like some of the explanatory text has been lost so this tutorial is less instructional than the previous version. The whole tutorial was intended to be a sequence so maybe we need to add back in an intro step statement for why each section is here. Line 43 maybe, line 75 absolutely since it explains WHY more changes are needed.